session/session_test.go

```diff
@@ -0,0 +1,176 @@
+package session
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestGetDeveloperPortalDataFromEnv(t *testing.T) {
+	// If the BITRISE_PORTAL_DATA_JSON env is already set (e.g for local testing the step),
+	// then we don't want to modify it during the test
+	var envAlreadyPresented bool
+	if os.Getenv("BITRISE_PORTAL_DATA_JSON") == "" {
+		if err := os.Setenv("BITRISE_PORTAL_DATA_JSON", dummyPortalDataJSON); err != nil {
+			t.Errorf("Failed to set BITRISE_PORTAL_DATA_JSON env to test getDeveloperPortalData() error = %v", err)
+		}
+	} else {
+		envAlreadyPresented = true
+	}
+
+	got, err := getDeveloperPortalData("", "")
```

Comment:
should not it always fail, because of: https://github.com/bitrise-io/steps-deploy-to-itunesconnect-deliver/pull/61/files#diff-928cb614ae092ed4c6f349f4c4c991e4R65

Reply:
Yes, It should.
This test will check the BITRISE_PORTAL_DATA_JSON parsing if:
- The BITRISE_PORTAL_DATA_JSON is not exported yet (we are not using it for locally debugging the step).

Then it will export it with dummy data and compare it with the result.

---

If the BITRISE_PORTAL_DATA_JSON is already exported, then:
-  It will not export the BITRISE_PORTAL_DATA_JSON (we don't want to override it)
- It will not compare it with the dummy data.